drivers: ethernet: stm32: Fix the init call order

juickar opened the pull request:
Fix the init call order for V2 API to be able to configure PTP as PTP initialization happens before iface initialization.

Fixes #96197 

This solution is the equivalent to the one used in the main branch at this commit: [ca6e25a](https://github.com/zephyrproject-rtos/zephyr/commit/ca6e25a03fa), but it can't be directly applied in the v4.2 branch, as the Ethernet driver is not split between v1 and v2 yet in this version.

juickar committed:
drivers: ethernet: stm32: Fix the init call order

Fix the init call order for V2 API to be able to configure PTP
as PTP initialization happens before iface initialization.

Signed-off-by: Julien Racki <julien.racki-ext@st.com>

maass-hamburg commented on drivers/ethernet/eth_stm32_hal.c:
```diff
@@ -930,7 +992,13 @@ static int eth_initialize(const struct device *dev)
 
-#endif /* !CONFIG_ETH_STM32_HAL_API_V1 */
+#elif defined(CONFIG_ETH_STM32_HAL_API_V2)
+	/* This function requires the Ethernet interface to be
+	 * properly initialized. In auto-negotiation mode, it reads the speed
+	 * and duplex settings to configure the driver accordingly.
+	 */
```
remove this comment, as it is wrong, since we switched to using the phy api completly

etienne-lms reviewed (commented):
Note there's an extra space char in your Signed-off-by tag (between your name and e-mail address) Compliance Checks CI test complains about.

juickar committed:
drivers: ethernet: stm32: Move the Ethernet API V1 init

Move the Ethernet API v1 Initialization in eth_init_api_v1()
helper function to match the V2 implementation.

Signed-off-by: Julien Racki <julien.racki-ext@st.com>

juickar commented on drivers/ethernet/eth_stm32_hal.c:
```diff
@@ -930,7 +992,13 @@ static int eth_initialize(const struct device *dev)
+	 * properly initialized. In auto-negotiation mode, it reads the speed
+	 * and duplex settings to configure the driver accordingly.
+	 */
+	eth_init_api_v2(dev);
```
I added a check and moved the v1 init in its own function

etienne-lms commented on drivers/ethernet/eth_stm32_hal.c:
```diff
@@ -851,7 +851,45 @@ static void RISAF_Config(void)
+
+	hal_ret = HAL_ETH_Init(heth);
+	if (hal_ret == HAL_TIMEOUT) {
+		/* HAL Init time out. This could be linked to */
+		/* a recoverable error. Log the issue and continue */
+		/* driver initialisation */
```
(feel free to discard my comment if you prefer)

juickar replied:
I also fixed it in the v2 function

maass-hamburg reviewed (approved):
Little nit, but fine for a fix, that will only be applied to a past version and not to main.
